convert2rhel/subscription.py

```diff
@@ -35,6 +35,17 @@ def subscribe_system():
         tool_opts.password = None
 
 
+def unregister_system():
+    """Unregister the system"""
+    loggerinst = logging.getLogger(__name__)
```

Comment:
Usually, loggers instantiated outside of the function scope. In your case, it will be always reinstating when unregister_system will be called. This https://docs.python.org/3/howto/logging-cookbook.html is useful

Reply:
Also useful to note that logging uses a singleton pattern, so each time you still have the same instance of it

Reply:
When you call `getLogger` you don't get a new instance but "a reference to the same logger object" as in the link you posted.
We used to take advantage of the singleton pattern in the past, we were simply calling e.g. `logger.log("warning", "Some warning")` from anywhere. But then John has changed it the way it is now, he was following https://fangpenlin.com/posts/2012/08/26/good-logging-practice-in-python/.
I'm also not in favor of getting the logger through `loggerinst = logging.getLogger(__name__)` in every function that wants to log. I'd be happy to get that changed.

Reply:
Oh, I am not sure which python version do you use, but since 2.7 disable_existing_loggers is True by default, so using module level instantiation is ok (moreover it is recommended in py2 and py3):
https://docs.python.org/2/howto/logging.html#advanced-logging-tutorial
https://docs.python.org/3/howto/logging.html#advanced-logging-tutorial

```
A good convention to use when naming loggers is to use a module-level logger, in each module which uses logging, named as follows:

logger = logging.getLogger(__name__)

This means that logger names track the package/module hierarchy, and it’s intuitively obvious where events are logged just from the logger name.
```

Reply:
Happy we know more about logging module now:) Never heard about disable exisiting loggers:)

Reply:
Currently, we need to be 2.6 and 2.7 compatible. Soon, we'll need to be also 3.6+ compatible. Getting the logger on the module level is fine with me.